File: src/main/java/pl/allegro/experiments/chi/chiserver/domain/experiments/ExperimentVariant.java
package pl.allegro.experiments.chi.chiserver.domain.experiments;

import com.google.common.base.Preconditions;
import joptsimple.internal.Strings;

import java.util.List;

public class ExperimentVariant {
    private final String name;
    private final List<Predicate> predicates;

    public ExperimentVariant(String name, List<Predicate> predicates) {
        Preconditions.checkArgument(!Strings.isNullOrEmpty(name));
        Preconditions.checkNotNull(predicates);
        this.name = name;
        this.predicates = predicates;
    }

    public String getName() {
        return name;
    }

    public List<Predicate> getPredicates() {
        return predicates;
    }

}

File: src/main/kotlin/pl/allegro/experiments/chi/chiserver/domain/experiments/ExperimentVariant.kt
package pl.allegro.experiments.chi.chiserver.domain.experiments

class ExperimentVariant(val name: String, val predicates: List<Predicate>) {

    init {
        require(name.isNotBlank()) { "empty name in ExperimentVariant" }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ExperimentVariant

        if (name != other.name) return false
        if (predicates != other.predicates) return false

        return true
    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + predicates.hashCode()
        return result
    }
}
